packages/neon_framework/lib/src/bloc/result.dart

```diff
@@ -151,10 +153,16 @@ class Result<T> {
 
   @override
   bool operator ==(Object other) =>
-      other is Result && other.isLoading == isLoading && other.data == data && other.error == error;
+      other is Result &&
+      other.isLoading == isLoading &&
+      other.data == data &&
+      other.error?.toString() == error?.toString();
+
+  @override
+  int get hashCode => Object.hash(data, error?.toString(), isLoading, isCached);
 
   @override
-  int get hashCode => Object.hash(data, error, isLoading, isCached);
+  String toString() => 'Result: { data: $data, error: $error, isLoading: $isLoading, isCached: $isCached }';
```

Comment:
```suggestion
  String toString() => 'Result(data: $data, error: $error, isLoading: $isLoading, isCached: $isCached)';
```
I prefer to have it like the constructor would look like.